lib/admin.rb

```diff
@@ -0,0 +1,149 @@
+require 'date'
+
+class Admin
+  attr_reader :rooms, :reservations
+
+  def initialize
+    @rooms = Array.new(20)
+      @rooms.length.times do |x|
+        @rooms[x] = x + 1
+      end
+
+    @reservations = []
+  end
+
+  def add_reservation(reservation, reservation_location: @reservations)
+    reservation_location << reservation
+  end
+
+  def request_reservation(start_date, end_date)
+    check_date(start_date, end_date)
+
+    id = reservations.length + 1
+    room = available_rooms(start_date, end_date).sample
+      raise ArgumentError, "no rooms available" if room.nil?
+
+    new_reservation = Reservation.new(id, room, start_date, end_date)
+
+    add_reservation(new_reservation)
+    return new_reservation
+  end
+
+  def request_block_reservation(number_of_rooms, start_date, end_date)
+    raise StandardError, "cannot reserve more than 5 rooms" if number_of_rooms > 5
+
+    check_date(start_date, end_date)
+    # individual reservations in block will be reached by going to block reservation first
+    # individual reservations with have different id's (1-5) depending on size of block
+    id = reservations.length + 1
+    room = available_rooms(start_date, end_date).sample(number_of_rooms)
+      raise StandardError, "not enough rooms available" if room.length < number_of_rooms
+
+    new_block = BlockReservation.new(id, room, start_date, end_date)
+
+    add_reservation(new_block)
+
+    return new_block
+  end
+  # param: reservation number
+  # returns - new reservation instance
+  def request_reservation_within_block(id, start_date, end_date)
+    #split this up into smaller methods possibly?
+    check_date(start_date, end_date)
+    check_id(id)
+
+    block_reservation = find_reservation_by_id(id)
+    # should I postfix these conditionals if it ends up exceeding the lines? -->
+    if Date.parse(start_date) != block_reservation.start_date
+      raise StandardError, "start date must match block start"
+    end
+    if Date.parse(end_date) != block_reservation.end_date
```

Comment:
I don't think this was explicitly made a requirement.

Reply:
I eliminated this so it will automatically make reservation in same time period as block reservation.